PVR API Extenstion: Get/SetRecordingLastPlayedPosition

Red-F opened the pull request:
This supersedes PR #513

Lars, as we discussed, this is the pvr api extension to get / set last watched position for a recording from / to the backend. I've added stubs in all the pvr clients that were present.

The api functions have been renamed to Get/SetRecordingLastPlayedPosition, as requested in the comments on PR# 513.

The calling of the new API functions have been wired into the existing code. 

GetRecordingLastPlayedPosition is called from xbmc/pvr/windows/GUIWindowPVRRecordings.cpp. The code will attempt to retrieve the last saved position from the back-end first, when successful a corresponding resume bookmark will be created in the video database since the recording may have been viewed on another device or with other software then the current XBMC. If there is no saved position on the back-end, or the functionality is not supported the code will try to fetch a resume bookmark from the video database just as it used to do.

SetRecordingLastPlayedPosition is called from xbmc/cores/dvdplayer/DVDPlayer.cpp, after much consideration. The reason for calling it there and not from xbmc/utils/SaveFileStateJob.h is that the time offset for the played file is NOT ALWAYS guaranteed to be filled in the SaveFileStateJob.

The code is working with the ForTheRecord add-on on Linux x86 and Windows (x86 and x64). I tried to build for OSX and iOS too, but the build is broken at the moment and after manually updating three of four of the makefiles I gave up. Will build and test on Linux x64 later this weekend.

Regards,
Fred

Commit:
PVR Api extended with lastplayed position management for recordings,
GetRecordingLastPlayedPosition will return the last watched position of
a recording in seconds (-1 on error/not implemented),
SetRecordingLastPlayedPosition will store the last watched position in
seconds for a recording.

Red-F commented:
Yep, working on Ubuntu amd64.

opdenkamp commented on xbmc/pvr/addons/PVRClient.cpp:
```diff
@@ -634,6 +634,59 @@ PVR_ERROR CPVRClient::RenameRecording(const CPVRRecording &recording)
+  }
+  catch (exception &e)
+  {
+    CLog::Log(LOGERROR, "PVRClient - %s - exception '%s' caught while trying to call RenameRecording() on addon '%s'. please contact the developer of this addon: %s",
```
copy+paste? :)

Red-F replied:
Definitely :)

opdenkamp commented:
please add a new bool for this feature in the add-on capabilities, so we don't needlessly call the add-on if we don't support this.

then you can also remove the "if (status != PVR_ERROR_NOT_IMPLEMENTED)" clauses.

Red-F commented:
Ok, one new bool coming up.

Commit:
Added bSupportsLastPlayedPosition to PVR_ADDON_CAPABILITIES. When true, this boolean indicates that the backend supports the store/retrieval operations of the last played position for a recording;
Corrected the reported method name in error message for SetRecordingLastPlayedPosition.

Red-F commented:
new capability boolean added, PVRClient accordingly updated (initialise and tests within new methods).
wrong method name in error message corrected.
all clients properly set the new capability boolean.

opdenkamp commented:
merged, thanks
